File: WebSite/EulerAList/pset22.cpp
#include <iostream>
#include <algorithm>
#include <sstream>
#include <vector>
#include <cmath>
#include <fstream>
#include <string>
using namespace std;

int main()
{
    vector<string> nameList;
    ifstream infile( "p022_names.txt" );
    while (infile)
    {
        string s;
        if (!getline(infile, s)) break;
        istringstream ss( s );
        while (ss)
        {
            string s;
            if (!getline( ss, s, ',' )) break;
            s.erase(remove(s.begin(), s.end(), '"'), s.end());
            nameList.push_back( s );
        }
    }
    infile.close();
    sort(nameList.begin(), nameList.end());
    long result = 0;
    for (int i = 1; i <= nameList.size(); i ++)
    {
        string s = nameList[i - 1];
        int sum = 0;
        for (char c : s)
        {
            sum += c - 'A' + 1;
        }
        result += sum * i;
    }
    cout << result << endl;
}

File: WebSite/EulerAList/pset22.py
nameList = []
infile = open("p022_names.txt", "r")
s = infile.read()
s = s.strip('"')
s = '"' + s + '"'
nameList = s.split(',')
infile.close
nameList = sorted(nameList)
result = 0
for i in range(1, len(nameList) + 1):
    name = nameList[i - 1]
    sum = 0
    for c in name:
        if ord(c) >= ord('A'):
            sum += ord(c) - ord('A') + 1  
    result += sum * i
print(result)
